src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,524 @@
+     *
+     * @return true if the provider is a subclass of the category.
+     */
+    private boolean isSubclass(Class<? extends RandomGenerator> category) {
+        return isSubclass(category, provider);
+    }
+
+    /**
+     * Return true if the provider is a subclass of the category.
+     *
+     * @param category Interface category, sub-interface of {@link RandomGenerator}.
+     * @param provider Provider that is being filtered.
+     *
+     * @return true if the provider is a subclass of the category.
+     */
+    private static boolean isSubclass(Class<? extends RandomGenerator> category,
+                                      Provider<? extends RandomGenerator> provider) {
+        return provider != null && category.isAssignableFrom(provider.type());
+    }
+
+    /**
+     * Returns the provider matching name and category.
+     *
+     * @param name      Name of RandomGenerator
+     * @param category  Interface category, sub-interface of {@link RandomGenerator}.
+     *
+     * @return A provider matching name and category.
+     *
+     * @throws IllegalArgumentException if provider is not a subclass of category.
+     */
+    private static Provider<? extends RandomGenerator> findProvider(String name,
+                                                                    Class<? extends RandomGenerator> category)
+            throws IllegalArgumentException {
+        Map<String, Provider<? extends RandomGenerator>> fm = getFactoryMap();
+        Provider<? extends RandomGenerator> provider = fm.get(name.toUpperCase());
+        if (!isSubclass(category, provider)) {
+            throw new IllegalArgumentException(name + " is an unknown random number generator");
+        }
+        return provider;
+    }
+
+    /**
+     * Returns a stream of matching Providers.
+     *
+     * @param category  Sub-interface of {@link RandomGenerator} to type check
+     * @param <T>       Sub-interface of {@link RandomGenerator} to produce
+     *
+     * @return Stream of matching Providers.
+     */
+    static <T extends RandomGenerator> Stream<RandomGeneratorFactory<T>> all(Class<? extends RandomGenerator> category) {
+        Map<String, Provider<? extends RandomGenerator>> fm = getFactoryMap();
+        return fm.values()
+                 .stream()
+                 .filter(p -> isSubclass(category, p))
+                 .map(RandomGeneratorFactory::new);
+    }
+
+    /**
+     * Returns a {@link RandomGenerator} that utilizes the {@code name} algorithm.
+     *
+     * @param name      Name of random number algorithm to use
+     * @param category  Sub-interface of {@link RandomGenerator} to type check
+     * @param <T>       Sub-interface of {@link RandomGenerator} to produce
+     *
+     * @return An instance of {@link RandomGenerator}
+     *
+     * @throws IllegalArgumentException when either the name or category is null
+     */
+    static <T> T of(String name, Class<? extends RandomGenerator> category)
```

Comment:
Same issue as above, T is not used as a constraint

Reply:
agree